sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/csv/CSVTable.scala

```diff
@@ -37,8 +37,13 @@ case class CSVTable(
     userSpecifiedSchema: Option[StructType],
     fallbackFileFormat: Class[_ <: FileFormat])
   extends FileTable(sparkSession, options, paths, userSpecifiedSchema) {
-  override def newScanBuilder(options: CaseInsensitiveStringMap): CSVScanBuilder =
-    CSVScanBuilder(sparkSession, fileIndex, schema, dataSchema, options)
+  override def newScanBuilder(options: CaseInsensitiveStringMap): CSVScanBuilder = {
+    val finalOptions = this.options.asCaseSensitiveMap().asScala.filterNot {
+      case (k, _) => options.containsKey(k)
+    }.toMap ++ options.asScala.toMap
+    CSVScanBuilder(sparkSession, fileIndex, schema, dataSchema,
+      new CaseInsensitiveStringMap(finalOptions.asJava))
```

Comment:
do other formats have the same issue?

Reply:
@cloud-fan orc/parquet/json/textTable do not combine the options of table and relation, while jdbcTable does merge them. I think we can add merging logic for all of them.